vscode/src/services/StatusBar.ts

```diff
@@ -9,6 +10,7 @@ import { FeedbackOptionItems } from './FeedbackOptions'
 interface StatusBarError {
     title: string
     description: string
+    errorType: string | RateLimitError
```

Comment:
```suggestion
    errorType: string
```

That should not be set to a class I assume? 🤔

Reply:
I changed this to `type StatusBarErrorName = 'auth' | 'RateLimitError'` for now (since these are the only two we use), since it felt a little less error-prone than just `string`.

Also pushed some fixes to the tests because the mocks didn't have the `hasError` impl.